src/coreclr/vm/jitinterface.cpp

```diff
@@ -11085,19 +11101,190 @@ PatchpointInfo* CEEJitInfo::getOSRInfo(unsigned* ilOffset)
     return result;
 }
 
-void CEEJitInfo::CompressDebugInfo()
+void CEEJitInfo::SetDebugInfo(PTR_BYTE pDebugInfo)
+{
+    ((CodeHeader*)m_CodeHeaderRW)->SetDebugInfo(pDebugInfo);
+}
+
+#ifdef FEATURE_INTERPRETER
+
+void CInterpreterJitInfo::allocMem(AllocMemArgs *pArgs)
+{
+    JIT_TO_EE_TRANSITION();
+
+    _ASSERTE(pArgs->coldCodeSize == 0);
+    _ASSERTE(pArgs->roDataSize == 0);
+
+    ULONG codeSize      = pArgs->hotCodeSize;
+    void **codeBlock    = &pArgs->hotCodeBlock;
+    void **codeBlockRW  = &pArgs->hotCodeBlockRW;
+    S_SIZE_T totalSize = S_SIZE_T(codeSize);
+
+    _ASSERTE(m_CodeHeader == 0 &&
+            // The jit-compiler sometimes tries to compile a method a second time
+            // if it failed the first time. In such a situation, m_CodeHeader may
+            // have already been assigned. Its OK to ignore this assert in such a
+            // situation - we will leak some memory, but that is acceptable
+            // since this should happen very rarely.
+            "Note that this may fire if the JITCompiler tries to recompile a method");
+    if( totalSize.IsOverflow() )
+    {
+        COMPlusThrowHR(CORJIT_OUTOFMEM);
+    }
+    if (ETW_EVENT_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_DOTNET_Context, MethodJitMemoryAllocatedForCode))
```

Comment:
Do we want to reuse the jit ETW events for interp? Or define new ones, or leave them out? Might be good to have an interpreter todo here if you think we'll want to update this later